### PR TITLE
NXP-24306: Refactor Plugin for 9.10 API

### PR DESCRIPTION
**Pull request by driverpt**

### CODE REVIEW

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/AclExcelLayoutBuilder.java [hunk 1]
```diff
@@ -26,9 +26,7 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.poi.hssf.util.HSSFColor;
-import org.apache.poi.ss.usermodel.CellStyle;
-import org.apache.poi.ss.usermodel.Font;
-import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.*;
```
Comment (nelsonsilva) [inline src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/AclExcelLayoutBuilder.java]: Avoid wildcard imports

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/seam/ExportGroupManagementActions.java [hunk 2]
```diff
@@ -36,6 +36,7 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.apache.poi.util.IOUtils;
```
Comment (nelsonsilva) [inline src/main/java/org/nuxeo/ecm/platform/groups/audit/seam/ExportGroupManagementActions.java]: Any advantage over our FileUtils ? Also we probably now have an unused import...

File: pom.xml [hunk 3]
```diff
@@ -128,6 +140,10 @@
       <artifactId>nuxeo-platform-userworkspace-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-directory-test</artifactId>
```
Comment (nelsonsilva) [inline pom.xml:145]: missing scope test scope?

File: pom.xml [hunk 4]
```diff
@@ -56,6 +56,18 @@
...
+      <artifactId>nuxeo-permissions</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-directory-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-directory</artifactId>
+    </dependency>
```
Comment (nelsonsilva) [inline pom.xml]: were we really missing all these compile time deps ?

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/excel/ExcelBuilder.java [hunk 5]
```diff
@@ -34,22 +34,11 @@
 import org.apache.poi.hssf.usermodel.HSSFPalette;
 import org.apache.poi.hssf.usermodel.HSSFPatriarch;
 import org.apache.poi.hssf.usermodel.HSSFWorkbook;
-import org.apache.poi.hssf.util.CellRangeAddress;
+
```
Comment (nelsonsilva) [inline src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/excel/ExcelBuilder.java]: Extra blank line

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/excel/ExcelBuilder.java [hunk 6]
```diff
@@ -34,22 +34,11 @@
...
-import org.apache.poi.ss.usermodel.CreationHelper;
-import org.apache.poi.ss.usermodel.Drawing;
-import org.apache.poi.ss.usermodel.Font;
-import org.apache.poi.ss.usermodel.Picture;
-import org.apache.poi.ss.usermodel.RichTextString;
-import org.apache.poi.ss.usermodel.Row;
-import org.apache.poi.ss.usermodel.Sheet;
-import org.apache.poi.ss.usermodel.Workbook;
-import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.apache.poi.ss.usermodel.*;
```
Comment (nelsonsilva) [inline src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/excel/ExcelBuilder.java]: Let's keep the explicit imports

**Review by nelsonsilva — APPROVED** [review]
LGTM with some comments.

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/seam/ExportGroupManagementActions.java (hunk 2 above)
Comment (driverpt) [reply src/main/java/org/nuxeo/ecm/platform/groups/audit/seam/ExportGroupManagementActions.java]: FileUtils is now a "Wrapper" to call IOUtils

File: pom.xml (hunk 3 above)
Comment (driverpt) [reply pom.xml:145]: You're right.

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/excel/ExcelBuilder.java (hunk 6 above)
Comment (driverpt) [reply src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/excel/ExcelBuilder.java]: IDE config. I'll fix it

File: pom.xml (hunk 4 above)
Comment (driverpt) [reply pom.xml]: I think it's better to explicitly add them.

File: pom.xml (hunk 3 above)
Comment (jcarsique) [reply pom.xml:145]: The `test` scope is the default for all `nuxeo-*-test` artifacts. Harmless to explicitly set it though.
```
<dependency>
  <groupId>org.nuxeo.ecm.platform</groupId>
  <artifactId>nuxeo-platform-directory-test</artifactId>
  <version>${nuxeo.services.version}</version>
  <scope>test</scope>
</dependency>
```

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/excel/ExcelBuilder.java (hunk 6 above)
Comment (jcarsique) [reply src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/excel/ExcelBuilder.java]: Which IDE? See https://github.com/nuxeo/nuxeo/tree/master/tools (complete or fix if not matching your own IDE)

**Comment by jcarsique** [comment]
:+1:

File: pom.xml (hunk 3 above)
Comment (driverpt) [reply pom.xml:145]: Given the amount of Plugins we have, i feel more confortable in leaving the ```<scope>``` there.
You never know when you refactor something and Plugins starts breaking.

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/seam/ExportGroupManagementActions.java (hunk 2 above)
Comment (efge) [reply src/main/java/org/nuxeo/ecm/platform/groups/audit/seam/ExportGroupManagementActions.java]: But don't use the `IOUtils` from Apache POI. Use `org.apache.commons.io.IOUtils`.

File: .gitignore [hunk 7]
```diff
@@ -16,4 +16,9 @@ log/
...
 *~
 *.xls
-*.xlsx
\ No newline at end of file
+*.xlsx
+.idea
+*.iml
+*.ipr
+*.iws
+*.log
```
Comment (efge) [inline .gitignore:24]: These should be part of your personal global `.gitignore`, not checked in.

File: src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/AclExcelLayoutBuilderMultiColumn.java [hunk 8]
```diff
@@ -113,36 +115,36 @@ protected void renderInit() {
...
 
         denyStyle = excel.newColoredCellStyle(ByteColor.RED);
-        denyStyle.setFillPattern(CellStyle.THIN_FORWARD_DIAG); // TODO:
-                                                               // generaliser
-                                                               // autres
-                                                               // cellules
+        denyStyle.setFillPattern(FillPatternType.THIN_FORWARD_DIAG);  // TODO:
+                                                                      // generaliser
+                                                                      // autres
+                                                                      // cellules
```
Comment (efge) [inline src/main/java/org/nuxeo/ecm/platform/groups/audit/service/acl/AclExcelLayoutBuilderMultiColumn.java]: As you're changing this line, rewrite the comment in English (generalize to other cells) and one one line.

File: src/test/resources/META-INF/MANIFEST.MF [hunk 9]
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-SymbolicName: nuxeo-groups-rights-audit-tests
```
Comment (efge) [inline src/test/resources/META-INF/MANIFEST.MF]: The convention we use for bundle symbolic names is of the form `org.nuxeo.ecm.platform.groups.audit.tests` (similar to Java package, can be simplified a bit if needed). Please also change the `Bundle-SymbolicName` of the non-test `MANIFEST.MF`.

**Review by efge — CHANGES_REQUESTED** [review]
Some changes

File: .gitignore (hunk 7 above)
Comment (nelsonsilva) [reply .gitignore:24]: We do have similar entries in our top level nuxeo/.gitignore ....

Comment (efge) [reply .gitignore:24]: Yeah but in my opinion they're wrong. Anything generated by personal tools (like IDEs and other editors) should be in a personal gitignore. Only extensions to ignore that are generated by build tools should be there (say, *.pyc, target, node_modules, ...). IMHO

Comment (nelsonsilva) [reply .gitignore:24]: I agree. Just mentioned 'cause I was doing the same remark and didn't after looking up our .gitignore

Comment (driverpt) [reply .gitignore:24]: I copied this from the main Nuxeo Repository. Personally, i prefer to create a `.gitignore` at https://www.gitignore.io/ but for consistency i used the one in nuxeo/nuxeo

**Review by efge — APPROVED** [review]

**Comment by efge** [comment]
👍

**Pull request merged**